src/Features/Core/Portable/ConvertIfToSwitch/AbstractConvertIfToSwitchCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,397 @@
+        //        : <pattern-expr> && <expr>                         // C#
+        //        | <expr0> is <pattern>                             // C#
+        //        | <expr0> is <type>                                // C#
+        //        | <expr0> == <const-expr>                          // C#, VB
+        //        | <expr0> <comparison-op> <const>                  //     VB
+        //        | ( <expr0> >= <const> | <const> <= <expr0> )
+        //           && ( <expr0> <= <const> | <const> >= <expr0> )  //     VB
+        //        | ( <expr0> <= <const> | <const> >= <expr0> )
+        //           && ( <expr0> >= <const> | <const> <= <expr0> )  //     VB
+        //
+        internal abstract class Analyzer
+        {
+            private SyntaxNode? _switchTargetExpression;
+            private readonly ISyntaxFactsService _syntaxFacts;
+
+            protected Analyzer(ISyntaxFactsService syntaxFacts)
+            {
+                _syntaxFacts = syntaxFacts;
+            }
+
+            public (ImmutableArray<AnalyzedSwitchSection>, SyntaxNode) AnalyzeIfStatementSequence(ReadOnlySpan<IOperation> operations)
+            {
+                var sections = ArrayBuilder<AnalyzedSwitchSection>.GetInstance();
+                if (!ParseIfStatementSequence(operations, sections, out var defaultBodyOpt))
+                {
+                    sections.Free();
+                    return default;
+                }
+
+                if (defaultBodyOpt is object)
+                {
+                    sections.Add(new AnalyzedSwitchSection(labels: default, defaultBodyOpt, defaultBodyOpt.Syntax));
+                }
+
+                Debug.Assert(_switchTargetExpression is object);
+                // UNDONE: Null-suppression should be removed once Assert is annotated
+                return (sections.ToImmutableAndFree(), _switchTargetExpression!);
+            }
+
+            // Tree to parse:
+            //
+            //    <if-statement-sequence>
+            //        : if (<section-expr>) { <unreachable-end-point> }, <if-statement-sequence>
+            //        : if (<section-expr>) { <unreachable-end-point> }, ( return | throw )
+            //        | <if-statement>
+            //
+            private bool ParseIfStatementSequence(ReadOnlySpan<IOperation> operations, ArrayBuilder<AnalyzedSwitchSection> sections, out IOperation? defaultBodyOpt)
+            {
+                if (operations.Length > 1 &&
+                    operations[0] is IConditionalOperation { WhenFalse: null } op &&
+                    HasUnreachableEndPoint(op.WhenTrue))
+                {
+                    if (!ParseIfStatement(op, sections, out defaultBodyOpt))
+                    {
+                        return false;
+                    }
+
+                    if (!ParseIfStatementSequence(operations.Slice(1), sections, out defaultBodyOpt))
+                    {
+                        var nextStatement = operations[1];
```

Comment:
i'm a little confused by the Slice(1) and the operations[1], does operations[1] refer to the 0th element of the slice?

Reply:
Slice passes the "rest" of the block, here we're only interested in the next statement to cover the "default" case, if there were no if/else i.e. `ParseIfStatementSequence` returned false.